src/ray/core_worker/core_worker.cc

```diff
@@ -454,17 +454,24 @@ CoreWorker::CoreWorker(const CoreWorkerOptions &options, const WorkerID &worker_
   periodical_runner_.RunFnPeriodically(
       [this] {
         const auto lost_objects = reference_counter_->FlushObjectsToRecover();
-        // Delete the objects from the in-memory store to indicate that they are not
-        // available. The object recovery manager will guarantee that a new value
-        // will eventually be stored for the objects (either an
-        // UnreconstructableError or a value reconstructed from lineage).
-        memory_store_->Delete(lost_objects);
-        for (const auto &object_id : lost_objects) {
-          // NOTE(swang): There is a race condition where this can return false if
-          // the reference went out of scope since the call to the ref counter to get
-          // the lost objects. It's okay to not mark the object as failed or recover
-          // the object since there are no reference holders.
-          RAY_UNUSED(object_recovery_manager_->RecoverObject(object_id));
+        if (!lost_objects.empty()) {
+          // Keep :info_message: in sync with LOG_PREFIX_INFO_MESSAGE in ray_constants.py.
+          RAY_LOG(ERROR)
+              << ":info_message: " << lost_objects.size()
```

Comment:
```suggestion
              << ":info_message:Reconstructing " << lost_objects.size() << " objects that were lost due to node failure.";
```

(to shorten).

Btw, is it possible to include the retry number, or is that already printed somewhere else?

Reply:
Hmm this log message wouldn't be completely accurate because we can still pin copies for those objects or they may not be reconstructable.

There is a separate INFO message once each individual task is retried, but that doesn't get piped to the driver right now. Added the attempt # to this message, though.

Reply:
How about we at least shorten then?

":info_message:Attempting to recovery N objects by re-submitting their tasks. To disable object reconstruction, set RAY_lineage_enabled=0."

Also, I think we should re-name the flag to make it a little more clean.

Reply:
K let me do the flag rename in a followup.